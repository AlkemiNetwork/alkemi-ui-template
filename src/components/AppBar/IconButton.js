import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import { withStyles } from "@material-ui/core/styles";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";

const styles = theme => ({
    menuButton: {
        marginLeft: 12,
        marginRight: 36
    },
    hide: {
        [theme.breakpoints.up("sm")]: {
            display: "none"
        }
    }
});

const AppbarButton = ({ classes, onClick, shifted }) => (
    <IconButton
        color="inherit"
        aria-label="open drawer"
        onClick={onClick}
        className={classNames(classes.menuButton, shifted && classes.hide)}
    >
        <MenuIcon />
    </IconButton>
);

AppbarButton.propTypes = {
    onClick: PropTypes.func.isRequired,
    classes: PropTypes.object.isRequired,
    shifted: PropTypes.bool
};

export default withStyles(styles)(AppbarButton);
