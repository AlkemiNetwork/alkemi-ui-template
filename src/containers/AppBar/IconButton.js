import { connect } from "react-redux";

import { openSidebar } from "../../redux/actions/ui";
import AppBarIconButton from "../../components/AppBar/IconButton";

const mapStateToProps = state => ({
    shifted: state.ui.sidebar.open
});

const mapDispatchToProps = {
    onClick: openSidebar
};

export default connect(mapStateToProps, mapDispatchToProps)(AppBarIconButton);
