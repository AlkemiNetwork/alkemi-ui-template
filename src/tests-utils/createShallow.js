import { shallow as _shallow } from "enzyme";

const createShallow = ({ dive }) => {
    const shallow = (node, options) => {
        let wrapper = _shallow(node, options);

        for (let i = 0; i < dive; i++) {
            if (wrapper.name().match(/^WithWidth(.+)$/)) {
                // Diving through a Meterial-UI WidthWith component can not be performed
                // by default because of React.Fragment used in withWidth
                // It is a known issue: https://github.com/airbnb/enzyme/issues/1213
                wrapper = wrapper
                    .dive()
                    .children()
                    .at(0);
            } else {
                wrapper = wrapper.dive();
            }
        }
        return wrapper;
    };

    return shallow;
};

export default createShallow;
