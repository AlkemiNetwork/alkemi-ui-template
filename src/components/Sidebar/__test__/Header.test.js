import React from "react";
import toJson from "enzyme-to-json";

import IconButton from "@material-ui/core/IconButton";
import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";

import SidebarHeader from "../Header";
import createShallow from "../../../tests-utils/createShallow";

describe("<SidebarHeader />", () => {
    let shallow;
    let wrapper;
    let props = {};

    beforeAll(() => {
        shallow = createShallow({ dive: 1 });
    });

    beforeEach(() => {
        props = {
            onClick: jest.fn()
        };
        wrapper = shallow(<SidebarHeader {...props} />);
    });

    it("renders 1 IconButton", () => {
        expect(wrapper.find(IconButton).length).toEqual(1);
    });

    it("renders 1 ChevronLeftIcon", () => {
        expect(wrapper.find(ChevronLeftIcon).length).toEqual(1);
    });

    it("IconButton button is functional", () => {
        wrapper
            .find(IconButton)
            .at(0)
            .simulate("click");
        expect(props.onClick.mock.calls.length).toEqual(1);
    });

    it("matches snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
