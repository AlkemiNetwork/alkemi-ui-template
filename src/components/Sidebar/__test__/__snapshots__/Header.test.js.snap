// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarHeader /> matches snapshot 1`] = `
<div
  className="SidebarHeader-drawerHeader-1"
>
  <WithStyles(IconButton)
    onClick={[MockFunction]}
  >
    <pure(ChevronLeftIcon) />
  </WithStyles(IconButton)>
</div>
`;
